invenio_accounts/models.py

```diff
@@ -90,6 +121,44 @@ class User(db.Model, Timestamp, UserMixin):
         "LoginInformation", back_populates="user", uselist=False, lazy="joined"
     )
 
+    @hybrid_property
+    def user_profile(self):
+        """Get the user profile."""
+        # NOTE: accessing this property requires an initialized app for config
+        if self._user_profile is None:
+            return None
+        elif not isinstance(self._user_profile, UserProfileDict):
+            self._user_profile = UserProfileDict(**self._user_profile)
+
+        return self._user_profile
+
+    @user_profile.setter
+    def user_profile(self, value):
+        """Set the user profile."""
+        if value is None:
+            self._user_profile = None
+        else:
+            self._user_profile = UserProfileDict(**value)
+
+    @hybrid_property
+    def user_preferences(self):
+        """Get the user preferences."""
+        # NOTE: accessing this property requires an initialized app for config
+        if self._preferences is None:
+            return None
+        elif not isinstance(self._preferences, UserPreferenceDict):
+            self._preferences = UserPreferenceDict(**self._preferences)
+
+        return self._preferences
+
+    @user_preferences.setter
+    def user_preferences(self, value):
+        """Set the user preferences."""
+        if value is None:
+            self._preferences = None
```

Comment:
maybe we should set this to `{}` rather than `None`?

Reply:
right now, the property setter makes sure that any value other than `None` is converted into a `UserPreferenceDict` (by calling the the latter's constructor with the deconstructed values), i.e. something like `user.user_profile = {"username": "max"}` (with a normal `dict`) is already fine
this is unnecessary overhead if the assigned value already is a `UserPreferenceDict`, but i think these assignments aren't happening so often that this would make a real difference...

if we rewrote the setter like the following (and the getter similarly):
```python
        if value is None:
            value = {}
        self._user_profile = UserProfileDict(**value)
```
we would have the benefit that we don't need `if user.user_profile is not None` checks in the code, before accessing its keys